src/marray.ts

```diff
@@ -194,7 +217,12 @@ export class MArray<T> {
   }
 
   update(index: number, val: T) {
+    const oldVal = tu.atIndex(this.$data, index).data;
```

Comment:
Its best to call `tu.atIndex(this.$data, index)` just once, otherwise we duplicate the retreival down the tree

Reply:
Done :+1: